### PR TITLE
[FEAT] Mining Moles

### PR DESCRIPTION
**Pull request by JacobSobolev**
# Description

Implementing the mining moles, based on issue #734 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
`yarn test`
image

**local testing:**
image
image
image

**mining with the master mole**
Animation

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by adamhannigan** [comment]
Hey @JacobSobolev All of the recipe data now lives on the blockchain and must be read from there. We will be adding these recipes onto the smart contracts early next week 😄

**Comment by JacobSobolev** [comment]
@adamhannigan 
I think it is finished now.

I tested the functionality locally and it works.
However, I wasn't able to test the following:
1. replenishing time
2. withdrawal prevention

File: src/features/goblins/bank/lib/bankUtils.ts [hunk 1]
```diff
@@ -70,6 +71,15 @@ export function canWithdraw({ item, game }: CanWithdrawArgs) {
     return !cropIsPlanted({ item: "Parsnip", game });
   }
 
+  // Make sure no trees are replenishing
```
Comment (ShiningMinions) [inline src/features/goblins/bank/lib/bankUtils.ts:74]: Not a big deal, but comment doesn't match the code.

Comment (JacobSobolev) [reply src/features/goblins/bank/lib/bankUtils.ts]: done

**Comment by adamhannigan** [comment]
Hey @JacobSobolev Thanks for the contribution ❤️ 

Will need to close this for now. Since it is so heavily architecture, API and smart contract driven the core team will need to take over

**Comment by JacobSobolev** [comment]
@adamhannigan sure thing, hope you will use some of my code at least :)

**Pull request closed without merging**